City-of-Helsinki/drupal-hdbt
# UHF-3504 contact card styling

# Contact card styling  [UHF-3504](https://helsinkisolutionoffice.atlassian.net/browse/UHF-3504)
Styles for contact card listing paragraph which can be added to standard pages upper and lower content areas as well as to tpr units upper and lower content areas

## What was done
* Styled the contact card listing paragraph according to [design](https://app.abstract.com/share/0c61faed-1984-437a-8dcd-d6bffcbb40aa?collectionLayerId=40779ec4-7b62-4118-adcd-eb4c0fd40049&mode=build&sha=9fd367fe8f4d18a0fcccce31c33df3c0192fe699) 
* Different from the design are telephone links and they are underlined on purpose.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh` or `make new`
* Update the HDBT theme and Platform config
    * run `make shell`
    * `composer require drupal/hdbt:dev-UHF-3504-contact-card-styling`
    * `composer require drupal/helfi_platform_config:dev-main` or switch platform config to main branch in local and fetch changed
    *`drush updb; drush-cr`

## How to test
* Test by creating or editing a standard page and add the new paragraph to the content area (should appear in both upper and lower content areas)
* Add the new paragraph and test that the styles work with and without images. Without images there should be a color line on the left side of the contact card info. A good guide line is to not mix contact cards with and without images but it also works that way
* Test that mobile styles work as well
* Check that if the photographer of the contact image is added, it should appear in the image alt text after the original alt text

NOTE: if image upload doesn't work, check docker desktop experimental settings and check both of the checkboxes.
